manifests/config.pp

```diff
@@ -47,7 +47,7 @@
     }
   }
 
-  if $::vault::install_method == 'archive' {
+  if $::vault::install_method == 'archive' or $::vault::manage_service_file {
```

Comment:
To maintain existing behaviour, the defaults should be:

 * manage_service_file = true when install_method == archive
 * manage_service_file = false when install_method == repo

There needs to be some conditional logic to achieve this, e.g. using `pick()`